lib/client/utils.js

```diff
@@ -51,12 +51,12 @@ self = module.exports = {
 
         switch (argType) {
         case "array":
-            if (!arg instanceof Array) {
+            if (!(arg instanceof Array)) {
                 invalidArg = true;
             }
             break;
         case "date":
-            if (!arg instanceof Date) {
+            if (!(arg instanceof Date)) {
```

Comment:
Wow, good catch - looks like that issue has been there forever. How'd you catch it?